File: 1894-merge-strings-alternately/merge-strings-alternately.cpp
class Solution { 
public:
    string mergeAlternately(string word1, string word2) {
        string res = "";
        int maxLength = max(word1.length(), word2.length());

        for(int i=0; i<maxLength; i++){
            if(i<word1.length()){
                res += word1[i];
            }
            if(i<word2.length()){
                res += word2[i];
            }
        }
        return res;
    }
};

File: 1894-merge-strings-alternately/merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder sb = new StringBuilder();
        int i = 0;
        while (word1.length() > i || word2.length() > i) {
            if (word1.length() > i){
                sb.append(word1.charAt(i));
            }
            if (word2.length() > i){
                sb.append(word2.charAt(i));
            }
            i++;
        }
        return sb.toString();
    }
}